Add faq page. (#895)
commercetools/commercetools-adyen-integration: docs/FAQ.md

# Frequently asked questions

### Can we use Adyen web components >= v5.0.0 with this integration ?

Yes, you could use the web components > v5.0.0. It's always better to keep your web component front end dependencies up to date with following latest Adyen releases.

In v5.0.0 (released in October 2021) Adyen introduced a simplified way of integrating Web Components, using a single API endpoint. The /sessions release is only an orchestration layer on top of existing functionalities.
This means that the complete checkout experience is still available and existing merchants will not have to change anything to their existing workflow. In fact, the 3 step /paymentMethods, /payments, /payments/details is still the way to go for more complex user flows.

> Adyen will be revisiting this topic in Q2 2022 and we will keep you updated on the potential upgrade regarding with /session endpoint usage with this integration.

Author: You can not use with integration now maybe in future if we add support for it with integration. as described this is not necessary for now until Adyen inform us to do a change.

> Adyen will be revisiting this topic in Q2 2022 and we will keep you updated on the potential upgrade regarding with /session endpoint usage with this integration.

Reviewer: But here you have this:

### Can we use Adyen web components >= v5.0.0 with this integration ?

Yes, you could use the web components > v5.0.0.

> Adyen will be revisiting this topic in Q2 2022 and we will keep you updated on the potential upgrade regarding with /session endpoint usage with this integration.

Author: I meant for session endpoint, but you could use the web components (js) and you should keep the deps up to date, wdyt how should we document this one. 
See [this](https://docs.adyen.com/online-payments/web-components/integrated-before-5-0-0) page it's mentioned as: "even as you update your Components version."

> Adyen will be revisiting this topic in Q2 2022 and we will keep you updated on the potential upgrade regarding with /session endpoint usage with this integration.

Reviewer: Hmm for me the update to new JS frontend version did not work, because I did not have the session thing (see our slack conversation with Adyen). That's why I think this statement is not correct. For now I would say `No, it's not possible to use web components >= 5.0.0. Please follow the documentation before v5.0.0: https://docs.adyen.com/online-payments/web-components/integrated-before-5-0-0`. You can leave the rest of your answer as it is.

Maybe you can also clarify this statement in the documentation with Adyen? Seems like it's not true.

> Adyen will be revisiting this topic in Q2 2022 and we will keep you updated on the potential upgrade regarding with /session endpoint usage with this integration.

Author: Removed the statement for now, will check later based on the discussion. c6d1ce4

### Does the integration support payment method X ?

Integration supports all [Adyen Web Component](https://docs.adyen.com/checkout/components-web) based payment methods. For a full list of payment methods please refer to [supported payment methods](https://docs.adyen.com/checkout/supported-payment-methods).
If you encounter any problems during your integration, please feel free to create a github issue.

> ### Shall I pass additional fields to payment requests, which additional fields that I need to pass with payment ?

Reviewer: ```suggestion
### Can I pass additional fields to payment requests?
```

Yes, you could include additional fields to payment such as [add risk management fields](https://docs.adyen.com/risk-management/configure-standard-risk-rules/required-risk-field-reference), [activate 3D Secure 2](https://docs.adyen.com/online-payments/3d-secure/native-3ds2/web-component#make-a-payment) or [allow recurring payments](https://docs.adyen.com/payment-methods/cards/web-component#create-a-token) based on the payment method that you use, for more details please check the note in [important section](../extension/docs/WebComponentsIntegrationGuide.md#step-5-make-a-payment) on our integration docs.

### Can I pass lineItems to makePayment requests ?

> Integration supports automatically adding lineItems from commercetools cart for klarna and affirm payments if the payment referenced in the cart, beside that lineItems and other required data can be provided by the users of the integration within the makePaymentRequest as answered in the above question.

Reviewer: ```suggestion
Integration supports automatically adding lineItems from commercetools cart for klarna and affirm payments if the payment is referenced in the cart, beside that lineItems and other required data can be provided by the users of the integration within the makePaymentRequest as answered in the above question. If lineItems field is provided, integration will skip adding lineItems and leave the provided lineItems.
```
